pyro/infer/reparam/transform.py

```diff
@@ -0,0 +1,31 @@
+import pyro
+import pyro.distributions as dist
+
+
+class TransformReparam:
+    """
+    Reparameterizer for
+    :class:`pyro.distributions.torch.TransformedDistribution` .
+
+    This is useful for transformed distributions with complex,
+    geometry-changing transforms, where the posterior has simple shape in
+    the space of ``base_dist``.
+
+    This reparameterization works only for latent variables, not likelihoods.
+    """
+    def __call__(self, name, fn, obs):
+        assert isinstance(fn, dist.TransformedDistribution)
+        assert obs is None
+
+        # Draw noise from the base distribution.
+        y = pyro.sample("{}_base".format(name), fn.base_dist)
+
+        # Differentiably transform.
```

Comment:
I think that we don't need to include `log_abs_det_jacobian` here. By reparameterizing (i.e. doing inference w.r.t. base value), we only need to include log_prob of base_dist.

Reply:
Fixed.